middleware.go

```
@@ -0,0 +1,60 @@
+package minio
+
+import (
+	"context"
+	"net/http"
+	"net/url"
+)
+
+// ExecutionContext provides context metadata about the S3 operation currently running.
+type ExecutionContext struct {
+	Method      string     // e.g. "GET", "PUT", "DELETE"
```

golangci-lint (2.12.2)

[error] File is not properly formatted

(gofumpt)